spatialos/UnrealGDK
# UNR-1076: Disconnection handling

#### Description
This change makes disconnection from SpatialOS trigger the vanilla Unreal codepath so that the necessary cleanup logic is run. As part of this, `USpatialNetDriver::OnDisconnected` and  `USpatialNetDriver::HandleOnDisconnected` have been removed in favour of using the native `UEngine::OnNetworkFailure` function. 

#### Release note
Bugfix: Worker disconnection flow is now handled by `UEngine::OnNetworkFailure` rather than the existing `OnDisconnection` callback, which has been removed

#### Tests
How did you test these changes prior to submitting this pull request?
* Forced timeouts in PIE and standalone game by killing SpatialOS during gameplay
* Attemped to launch game from PIE when SpatialOS is not running
 
How can this be verified by QA?
* The same steps as above can be followed - ie, attempt to launch the game in PIE without SpatialOS running locally, and killing the runtime process while connected to the game. Ensure that the logs eventually print out `NetworkFailure` messages such as 
--- LogNet: Warning: Network Failure: GameNetDriver[PendingConnectionFailure]: Failed to connect to the receptionist...
--- LogNet: NetworkFailure: PendingConnectionFailure, Error: Failed to connect to the receptionist...

#### Documentation
* In-line comments regarding error-code mapping
